CAIPs/caip-draft_cacao.md

```diff
@@ -0,0 +1,176 @@
+signing operation as an [IPLD](https://ipld.io)-based object capability (OCAP).
+As we expect other blockchains to follow a path similar to Ethereum, CAIP seems to be the best place for such a proposal.
+
+## Motivation
+
+"Sign-in with Ethereum" is a way for a user to authenticate into a service, and provide authorization. In essence, it is a signature of a well-formed payload, that can be read by a human as well as a machine.
+We could see this as a stepping point for a _rich_ capability-based authorization system.
+In order to do this, we would like to have a standardized IPLD-based representation of the payload and the signature, that together comprise a capability.
+
+## Specification
+
+### Container format
+
+We start construction with declaring a container format, that represents a signed payload.
+It should contain meta-information, payload and signatures. For reference let's call such container _CACAO_:
+
+```
+type CACAO struct {
+  h Header // container meta-information
+  p Payload // payload
+  s Signature // signature, single
+}
+```
+
+Header uniquely identifies signature verification process:
+
+```
+type Header struct {
+  t String // specifies signature verification algorithm and format of the payload
+}
+```
+
+For now we expect this to be either "eip4361-eip191" or "eip4361-eip1271". For both formats the payload structure must be presented as follows:
+
+```
+type Payload struct {
+  domain String // =domain
+  iss String // = DID pkh
+  aud String // =uri
+  version String
+  nonce String
+  iat Int // Unix timestamp =issued-at
+  nbf optional Int // Unix timestamp =not-before
+  exp optional Int // Unix timestamp = expiration-time
+  statement optional String // =statement
+  requestId optional String // =request-id
+  resources optional [ String ] // =resources as URIs
+}
+```
+
+It is important to note, that issuer here is [did:pkh](https://github.com/spruceid/ssi/blob/main/did-pkh/did-pkh-method-draft.md), which includes both blockchain address and blockchain network information.
+
+The difference is how we do signature verification.
+
+The signature in essence is just bytes, but we also would like to provide additional meta-information at a later time.
+
+```
+type Signature struct {
+  m optional SignatureMeta // For future extension
+  s Bytes
+}
+
+type SignatureMeta struct {
+}
+```
+
+### Signature Verification
+
+We reconstruct the EIP4361 payload as follows:
+
+```
+{.p.domain} wants you to sign in with your Ethereum account:
+{.p.iss[address]}
+
+{.p.statement}
+
+URI: {.p.aud}
+Version: {.p.version}
+Chain ID: {.p.iss[chainId]}
+Nonce: {.p.nonce}
+Issued At: {.p.iat}
+Resources:
+- {.p.resources[0]}
+- {.p.resources[1]}
+...
+- {.p.resources[n]}
+```
+
+Then using [EcdsaSecp256k1RecoveryMethod2020](https://identity.foundation/EcdsaSecp256k1RecoverySignature2020/) we verify the signature of the payload above, if type is "eip4361-eip191".
```

Comment:
EcdsaSecp256k1RecoveryMethod2020 is a linked data proof suite, for signing/verifying JSON-LD/RDF data. But the payload data here is a string, for the user/wallet to hash and sign according to `EIP-191`/`personal_sign`.
```suggestion
We then verify the signature of the payload above, according to [EIP-191](https://eips.ethereum.org/EIPS/eip-191) if type is "eip4361-eip191".
```

Reply:
Thank you @clehner !